rustprocessor/src/handler/payload.rs

```diff
@@ -0,0 +1,149 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * -----------------------------------------------------------------------------
+ */
+
+use std::str;
+use std::fmt;
+
+use sawtooth_sdk::processor::handler::ApplyError;
+
+#[derive(Copy, Clone)]
+pub enum Action {
+    Deposit,
+    Withdraw,
+    Balance,
+    Transfer,
+}
+
+impl fmt::Display for Action {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(
+            f,
+            "{}",
+            match *self {
+                Action::Deposit => "Action::Deposit",
+                Action::Withdraw => "Action::Withdraw",
+                Action::Balance => "Action::Balance",
+                Action::Transfer => "Action::Transfer",
+            }
+        )
+    }
+}
+
+
+pub struct SwPayload {
+    action: Action,    
+    value: u32,
+    beneficiary_pubkey: Option<String>,
+}
+
+impl SwPayload {
+
+    pub fn new(payload_data: &[u8]) -> Result<Option<SwPayload>, ApplyError> {
+    
+        let payload_string = match str::from_utf8(&payload_data) {
+            Ok(s) => s,
+            Err(_) => {
+                return Err(ApplyError::InvalidTransaction(String::from(
+                    "Invalid payload serialization",
+                )))
+            }
+        };
+
+        //SimpleWallet payload is constructed as comma seperated items
+        let items: Vec<&str> = payload_string.split(",").collect();
+
+        if items.len() < 2 {
+            return Err(ApplyError::InvalidTransaction(String::from(
+                "Payload must have atleast 1 commas",
+            )));
+        }
+        
+        if items.len() > 3 {
+            return Err(ApplyError::InvalidTransaction(String::from(
+                "Payload must have atmost 2 commas",
+            )));
+        }
+                
+        let (action, value) = (items[0], items[1]);
+        
+        if action.is_empty() {
+            return Err(ApplyError::InvalidTransaction(String::from(
+                "Action is required",
+            )));
+        }
+             
+        
+        let action = match action {
+                            "deposit" => Action::Deposit,
+                            "withdraw" => Action::Withdraw,
+                            "balance" => Action::Balance,
+                            "transfer" => Action::Transfer,
+                            _ => {
+                                return Err(ApplyError::InvalidTransaction(String::from(
+                                    "Invalid Action",
+                                )))
+                          }
+        };
+ 
+         
+        let value: u32 = match value.parse() {
+            Ok(num) => num,
+            Err(_) => {
+                return Err(ApplyError::InvalidTransaction(String::from(
+                    "Space must be an integer",
```

Comment:
I don't understand this message: "Space must be an integer"
Do you mean the integer is missing (empty)? Or what?

Reply:
changed message and submitted new commit